### PR TITLE
Validate performance test keys

### LINKED ISSUE #6823: Ledger API Test Tool runs conformance tests if performance test names are not valid
If an invalid key is passed to `--perf-tests`, instead of quitting the Ledger API Test Tool runs the full suite of conformance tests, which is not the behavior the user would expect.

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
Fixes #6823

changelog_begin
[Integration Kit] Fixed a bug in the Ledger API test tool that caused
the full conformance test suite to be run when trying to run performance
tests but using a wrong name. See https://github.com/digital-asset/daml/issues/6823
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/Cli.scala [hunk 1]
```diff
@@ -146,6 +146,12 @@ object Cli {
       .text("""A comma-separated list of inclusion prefixes. If not specified, all default tests are included. If specified, only tests that match at least one of the given inclusion prefixes (and none of the given exclusion prefixes) will be run. Can be specified multiple times, i.e. `--include=a,b` is the same as `--include=a --include=b`.""")
 
     opt[Seq[String]]("perf-tests")
+      .validate(tests =>
+        if (tests.forall(Tests.PerformanceTestsKeySet)) {
+          success
+        } else {
+          failure("Invalid performance test name, use --list to see valid performance test names")
```
Comment (SamirTalwar-DA) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/Cli.scala:153]:
```suggestion
          failure("Invalid performance test name. Use `--list` to see valid names.")
```

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/Cli.scala [hunk 2]
```diff
@@ -146,6 +146,12 @@ object Cli {
...
+      })
```
Comment (SamirTalwar-DA) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/Cli.scala:154]: Or perhaps something like this:

```suggestion
      .validate(tests =>
        tests.find(Tests.PerformanceTestsKeySet) match {
          None =>
            success
          Some(test) =>
            failure(s"Invalid performance test name: '$test'. Use `--list` to see valid names.")
      })
```

Comment (stefanobaghino-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/Cli.scala]: I'll use this suggestion, but I believe this logic is the opposite of what we want. :wink:

Comment (stefanobaghino-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/Cli.scala]: Addressed by https://github.com/digital-asset/daml/pull/6826/commits/ed7c37fb0a12564901f8bbdf63d080b35a523d49
Thanks for the suggestion. :bow:

Comment (SamirTalwar-DA) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/Cli.scala]: Eh, you figured it out. 😛

**Pull request merged**